src/ProjNavigator/sources_form.cpp

```diff
@@ -264,15 +264,19 @@ void SourcesForm::SlotAddFile() {
 }
 
 void SourcesForm::SlotOpenFile() {
-  QTreeWidgetItem *item = m_treeSrcHierachy->currentItem();
-  if (item == nullptr) {
-    return;
+  // Get selected file names
+  for (auto item : m_treeSrcHierachy->selectedItems()) {
+    // Using WhatsThis role to ensure selection is a file
+    // OpenFile quietly aborts on bad filenames, so this check could be removed
+    // if a desired type is being missing
+    if (item->data(0, Qt::WhatsThisPropertyRole)
+            .toString()
+            .contains("fileitem")) {
```

Comment:
I don't see using "fileitem" anywhere. Where is it come from?

Reply:
"fileitem" is part of a string used in the WhatsThisPropertyRole for each file type. You can find the defines here https://github.com/os-fpga/FOEDAG/blob/6b95d2bb9dacd37585c0659e03c5c7c7827bf9bc/src/ProjNavigator/sources_form.h#L16. So we are basically targeting SRC_TREE_DESIGN_FILE_ITEM, SRC_TREE_CONSTR_FILE_ITEM, SRC_TREE_SIM_FILE_ITEM & SRC_TREE_IP_FILE_ITEM with this.